allennlp/data/dataset_readers/universal_dependencies.py

```diff
@@ -49,6 +49,10 @@ def _read(self, file_path: str):
             logger.info("Reading UD instances from conllu dataset at: %s", file_path)
 
             for annotation in  lazy_parse(conllu_file.read()):
+                # Conllu annotations can have implicit word references that have been
+                # filled in automatically and have no head, so we remove them.
```

Comment:
This is called ["ellipsis"](https://en.wikipedia.org/wiki/Ellipsis_(linguistics)), and it took me a bit to figure out what you meant by "implicit word references".  Here's a stab at a comment that's more intuitive to me:  "CoNLLU annotations sometimes add back in words that have been elided in the original sentence; we remove these, as we're just predicting dependencies for the original sentence."

It's not obvious to me how `x["id"] is None` maps to words that have been added, though.